### DISCUSSION TITLE
vcpkg_fixup_pkgconfig does not fix includedir of debug library

### DISCUSSION THREAD

**Discussion by maz-1** [microsoft/vcpkg#24297] [Ideas]
**Describe the bug**
vcpkg_fixup_pkgconfig does not fix includedir of debug library.
some packages (like gdal) fix this by explicitly replacing `${prefix}/include` with `${prefix}/../include`. I think this should be done by vcpkg_fixup_pkgconfig instead of port maintainer

**Environment**
- OS: Windows

**Expected behavior**
vcpkg_fixup_pkgconfig takes care of everything, no more need to explicitly replace.

**Comment by dg0yt** [comment]
vcpkg_fixup_pkgconfig does attempt to fix the include dir of the debug variant. Key spot:
https://github.com/microsoft/vcpkg/blob/af2287382b1991dbdcb7e5112d236f3323b9dd7a/scripts/cmake/vcpkg_fixup_pkgconfig.cmake#L155-L160

> some packages (like gdal) fix this by explicitly replacing `${prefix}/include` with `${prefix}/../include`. I think this should be done by vcpkg_fixup_pkgconfig instead of port maintainer

The fixup depends on matching certain patterns. There will always be some packages which use unique patterns. So the port maintainers need to take care of exceptions.

But of course, concrete proposals for improvement are welcome.

**Comment by maz-1** [comment]
OK, sounds like we need a function that can convert all the variables into absolute path and work out the hierarchical relationship.
